src/api/posts.ts

```diff
@@ -0,0 +1,25 @@
+import { Post } from '../react-app-env';
+
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getPosts(userId: string): Promise<Post[]> {
+  const response = await fetch(`${BASE_URL}/posts?userId=${userId}`);
+
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
+}
```

Comment:
as userId can be null, use (userId: string | null)

add param to get all posts depends on userId
const userIdParam = userId && userId !== "ALL" ? "userId=" + userId : "";

modify request
const response = await fetch(`${BASE_URL}/posts?${userIdParam}`);